### ISSUE TITLE
[BUG] Problem with AGS (Sidebar/Clock does nothing on click waybar)

### ISSUE THREAD

**Issue by mrjxtr** [mylinuxforwork/dotfiles#492]
**Describe the bug**
Ever since this ags bug happened where when you click the ML4W logo or the time, it does nothing, I have tried multiple ways to make it work, including reinstalling ags, updating to a newer version of ML4w, installing a new ML4W dotfile in a new folder. But when I click them, still nothing happens.

In addition to this, using the welcome app to "update/reinstall ml4w dotfiles" no longer works. It just opens the terminal and closes. So I have to do it manually everytime using `yay -Syu` then running `ml4w-hyprland-setup`.

What I did so far:
- update everything including ml4w using `yay -Syu` then running `ml4w-hyprland-setup`
- reinstalled ags with `yay -S aylurs-gtk-shell`
- installed ml4w dotfiles on another folder ~/dotfiles_new

I also found the waybar modules.json with contains this the code below. (I am not sure if this helps but I tried)
- There is not ~/.config/ml4w/apps/ directory when I tried to cd into it.
- When I try to run the commands `ags toggle sidebar` or `ags toggle calendar` it will return this:

```
❯ ags toggle sidebar
error: there is no "astal" instance runnning

❯ ags toggle calendar
error: there is no "astal" instance runnning
```

modules.json
```
  // ML4W Welcome App
  "custom/ml4w-welcome": {
    "on-click": "ags toggle sidebar",
    "on-click-right": "sleep 0.1 && ~/.config/ml4w/apps/ML4W_Welcome-x86_64.AppImage",
    "format": " ",
    "tooltip-format": "Open Sidebar"
  },

  // Clock
  "clock": {
        "format": "{:%H:%M %a %y.%Om.%Od}",
    "on-click": "ags toggle calendar",
    "tooltip": false
  },

```

**Distribution (please complete the following information):**
 - Arch Linux

**Additional context**
I have ml4w-hyprland-git 2.9.6.7.r89.g96f8ac1-1 (+0 0.00) (Installed)

**Comment by mylinuxforwork** [comment]
@mrjxtr I will release 2.9.7 later today. Please test with the new version. It's important that ags 2.0.x and astal are installed correctly. Please check the ags github page.

**Comment by mrjxtr** [comment]
> @mrjxtr I will release 2.9.7 later today. Please test with the new version. It's important that ags 2.0.x and astal are installed correctly. Please check the ags github page.

I will, thanks. Hopefully it fixes my issue 🙏🏼

and yes, I do have ags 2.0.x and astal installed

**Comment by mrjxtr** [comment]
@mylinuxforwork I use `ml4w-hyprland-setup` after all updates, sidebar still not working, I use the update/reinstall script from the ml4w welcome app to update, but still not working :(

**Comment by mylinuxforwork** [comment]
@mrjxtr Your installation of ags and astal failed. Are you on Arch?

**Comment by mrjxtr** [comment]
> @mrjxtr Your installation of ags and astal failed. Are you on Arch?

@mylinuxforwork yes, I am on arch... That's interesting because all of the checks it did says 'already installed'... How do you propose me to fix this? 🥲

**Comment by mylinuxforwork** [comment]
> > @mrjxtr Your installation of ags and astal failed. Are you on Arch?
> 
> @mylinuxforwork yes, I am on arch... That's interesting because all of the checks it did says 'already installed'... How do you propose me to fix this? 🥲

Can you just try this https://aylur.github.io/ags/guide/quick-start.html

**Comment by mrjxtr** [comment]
@mylinuxforwork 

I did the three commands listed on that page: 

- It replaced `aylurs-gtk-shell` with `aylurs-gtk-shell-git`
- After then I did `ags init` and `ags run` and here are the results:

```
❯ ags init
error: could not initialize: /home/mrjxtr/.config/ags already exists

❯ ags run
error: no such file or directory: "/home/mrjxtr/.config/ags/app"
tip: valid names are:  "app.js" "app.ts" "app.jsx" "app.tsx"
```

**Comment by mylinuxforwork** [comment]
> @mylinuxforwork
> 
> I did the three commands listed on that page:
> 
> * It replaced `aylurs-gtk-shell` with `aylurs-gtk-shell-git`
> * After then I did `ags init` and `ags run` and here are the results:
> 
> ```
> ❯ ags init
> error: could not initialize: /home/mrjxtr/.config/ags already exists
> 
> ❯ ags run
> error: no such file or directory: "/home/mrjxtr/.config/ags/app"
> tip: valid names are:  "app.js" "app.ts" "app.jsx" "app.tsx"
> ```
> 
> Also tried to install astal and looks like it is already installed:
> 
> ```
> ❯ yay -S libastal-io-git libastal-git
> AUR Dependency (2): libastal-git-r540.c5eba25-2, libastal-io-git-r438.c84d710-1
> Sync Make Dependency (1): vala-0.56.17-1
> :: (1/2) Downloaded PKGBUILD: libastal-io-git
> :: (2/2) Downloaded PKGBUILD: libastal-git
>   2 libastal-git                     (Installed) (Build Files Exist)
>   1 libastal-io-git                  (Installed) (Build Files Exist)
> ==> Packages to cleanBuild?
> ==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
> ==> ^C
> 
> ❯ yay -S libastal-meta
> AUR Dependency (1): libastal-meta-1-8
> :: (1/1) Downloaded PKGBUILD: libastal-meta
>   1 libastal-meta                    (Installed) (Build Files Exist)
> ==> Packages to cleanBuild?
> ==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
> ==> ^C
> ```
> 
> The only notable error I can find which I also mentioned previously is this :
> 
> ```
> error: no such file or directory: "/home/mrjxtr/.config/ags/app"
> ```
> 
> I also don't have this `~/.config/ml4w/apps/ML4W_Welcome-x86_64.AppImage` which was included in the `modules.json` for waybar that I mentioned in the first comment

OK. You’re not on the latest ml4w dotfiles. Please run yay -S ml4w-hyprland and then the command ml4w-hyprland-setup

This should install 2.9.7.0 and then tey again.

**Comment by mrjxtr** [comment]
@mylinuxforwork

Done running `yay -S ml4w-hyprland` and then the command `ml4w-hyprland-setup`

now I have this:

```
❯ ags init
error: could not initialize: /home/mrjxtr/.config/ags already exists
❯ ags run
✘ [ERROR] Could not resolve "./overrides"

    ../../../../../usr/share/astal/gjs/index.ts:1:7:
      1 │ import "./overrides"
        ╵        ~~~~~~~~~~~~~

✘ [ERROR] Could not resolve "./overrides"

    ../../../../../usr/share/astal/gjs/_app.ts:1:7:
      1 │ import "./overrides"
        ╵        ~~~~~~~~~~~~~
```
